sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/SecretClientJavaDocCodeSnippets.java

```diff
@@ -138,24 +138,25 @@ public void updateSecretWithResponseCodeSnippets() {
     public void deleteSecretCodeSnippets() throws InterruptedException {
         SecretClient secretClient = getSecretClient();
         // BEGIN: com.azure.security.keyvault.secretclient.deleteSecret#string
-        Poller<DeletedSecret, Void> deletedSecretPoller = secretClient.beginDeleteSecret("secretName");
-
-        while (deletedSecretPoller.getStatus() != PollResponse.OperationStatus.IN_PROGRESS
-            && !deletedSecretPoller.isComplete()) {
-            System.out.println(deletedSecretPoller.getStatus().toString());
-            Thread.sleep(2000);
-        }
-
-        // Deleted Secret is accessible as soon as polling begins
-        DeletedSecret deletedSecret = deletedSecretPoller.getLastPollResponse().getValue();
-        System.out.println("Deleted Date  %s" + deletedSecret.getDeletedOn().toString());
-        System.out.printf("Deleted Secret's Recovery Id %s", deletedSecret.getRecoveryId());
-
-        // Secret is being deleted on server.
-        while (!deletedSecretPoller.isComplete()) {
-            System.out.println(deletedSecretPoller.getStatus().toString());
-            Thread.sleep(2000);
-        }
+        Poller<DeletedSecret, Void> poller = secretClient.beginDeleteSecret("secretName");
+
+        // Listen to the poller for each poll response and block synchronously until the delete operation completes.
+        poller.getObserver()
+            .map(pollResponse -> {
+                System.out.println("Current status: " + pollResponse.getStatus());
+
+                // Deleted secret can be accessed as soon as polling is in progress.
+                if (pollResponse.getStatus() != PollResponse.OperationStatus.NOT_STARTED) {
+                    DeletedSecret deletedSecret = pollResponse.getValue();
+                    System.out.println("Deleted date: " + deletedSecret.getDeletedOn().toString());
+                    System.out.printf("Deleted Secret's Recovery id: %s%n", deletedSecret.getRecoveryId());
```

Comment:
On Sync client, 
We now have a sync poller coming in.
This will create conflicts.
Plus showcasing reactor style on sync side, doesn't feel right.

Reply:
This snippet is shared in the async client side.

Reply:
I can move the snippet to the async secret client.

Reply:
Actually, I see the problem, the snippets are named after the class names, but these names are secretclient rather than secretasyncclient. I can revert them so it doesn't conflict the changes.